### PR TITLE
feat(forum): improve thread viewing ux on desktop and mobile

### PR DESCRIPTION
**Pull request by wescopeland**
This PR makes numerous improvements to the site's forum thread viewing UX. The changes are most impactful on mobile, but are felt across all breakpoints. Some things here are fixes, others are new features.

**Why?**

Currently, it can be very difficult to read threads on mobile. Post meta is cramped, and things generally feel busy. As I started going down the rabbit hole of improving this, the scope of what I felt could be improved continued growing and growing, all culminating into this PR.

**Summary of Changes**

* There is no longer a "dead zone" on the left-hand side of the screen when viewing forum threads on mobile.
* Post metadata no longer clips random letters of the word "Edited" on mobile.
* On mobile, user metadata appears at the top of each post. On desktop, it appears at the left.
* User meta now contains: avatar, username, user role name if something other than "Registered", post count, and join date.
* Banned users no longer have their profiles linked or their avatars shown. All that is shown now is a striked-through username and their role: "Banned".
* On every thread there is now an "OP" designator tagged to the original post and any replies made by the original poster.
* The copy link buttons now also show the current post count to help users orient themselves to where they are in the thread, especially longer threads.

**Technical Implementation**

* This is implemented with a collection of Blade components that reside in _resources/views/community/components/forum_.

## Videos

https://user-images.githubusercontent.com/3984985/235366341-2291a78f-4ac1-45d8-99bf-04bb54a68e4e.mp4

https://user-images.githubusercontent.com/3984985/235366342-3f5ef0a4-b6be-40dc-a197-a42d85fc3c43.mp4

## Comparisons

**Desktop View, OLD**

Desktop View Old

**Desktop View, NEW**

Desktop View New

---

**Mobile View, OLD**

Mobile View Old

**Mobile View, NEW**

Mobile View New

---

**User Meta**

User Meta New

Banned Users Desktop New

---

**OP Designation**

Post Meta New

---

**Authorization Tools**

Moderation Actions New

---

**Highlights, OLD**

Highlight Old

**Highlights, NEW**

Highlight New

### CODE REVIEW

File: resources/css/breadcrumb.css [hunk 1]
```diff
@@ -14,4 +14,5 @@
 
 .navpath > * {
   display: inline-block;
+  @apply leading-4;
```
Comment (wescopeland) [inline resources/css/breadcrumb.css:17]: For consistency, apply the game page's line-height change site wide. This greatly improves the readability of some forum thread breadcrumbs that contain emojis, such as AOTW.

File: app_legacy/Helpers/database/forum.php [hunk 2]
```diff
@@ -145,7 +145,8 @@ function getTopicComments(int $topicID, int $offset, int $count, ?int &$maxCount
         $maxCountOut = (int) $data['COUNT(*)'];
     }
 
-    $query = "SELECT ftc.ID, ftc.ForumTopicID, ftc.Payload, ftc.Author, ftc.AuthorID, ftc.DateCreated, ftc.DateModified, ftc.Authorised, ua.RAPoints
+    $query = "SELECT ftc.ID, ftc.ForumTopicID, ftc.Payload, ftc.Author, ftc.AuthorID, ftc.DateCreated, ftc.DateModified, ftc.Authorised, ua.RAPoints, ua.Created AS AuthorJoined, ua.Permissions as AuthorPermissions,
+                (SELECT COUNT(*) FROM ForumTopicComment WHERE AuthorID = ftc.AuthorID) AS AuthorPostCount
```
Comment (Jamiras) [inline app_legacy/Helpers/database/forum.php:149]: This adds a not insignificant amount of overhead to the query. I'm not sure if it's enough to ditch it, or even if it's worth it in the first place. 

image
image

Maybe we can revisit with denormalized data later?

File: public/viewtopic.php [hunk 3]
```diff
@@ -150,114 +149,43 @@
...
-                // Allow, only if this is MY comment (disclaimer: unofficial), or if I'm admin (disclaimer: unofficial, verify user?)
-                if ($permissions >= Permissions::Admin) {
-                    // Allow with disclaimer
-                    $showDisclaimer = true;
-                    $showAuthoriseTools = true;
-                } elseif ($nextCommentAuthor == $user) {
-                    // Allow with disclaimer
-                    $showDisclaimer = true;
-                } else {
-                    continue;    // Ignore this comment for the rest
```
Comment (Jamiras) [inline public/viewtopic.php:196]: You're no longer excluding unauthorized posts (they should only be visible for the author and admins).

Comment (Jamiras) [reply public/viewtopic.php:196]: Fixing this will probably affect the numbering on the links.

File: app_legacy/Helpers/database/forum.php (hunk 2 above)
Comment (wescopeland) [reply app_legacy/Helpers/database/forum.php]: You're right, this should wait for denormalized data. I've stripped post count but left the user role. I will try to get in a better habit of testing the query performance impact of these changes before putting them in PRs. Addressed in https://github.com/RetroAchievements/RAWeb/pull/1534/commits/f708977a67a1788471ac2aee7c6ecc72175d27fb.

**Review by Jamiras — APPROVED** [review]

**Comment by Jamiras** [comment]
Tagging @luchaos for input on UX layout and blade conversion.

**Comment by Xeinok-Pals** [comment]
zebra that bad boi upppp

**Comment by wescopeland** [comment]
@luchaos @Jamiras This branch has been updated to resolve all outstanding conflicts with V3.

**Review by Jamiras — APPROVED** [review]

**Pull request merged**